src/tokens/slice.ts

```diff
@@ -149,19 +145,22 @@ const slice = createSlice({
       loading: false,
       error: true,
     }),
-    importToken: (state, action: PayloadAction<BaseToken>) => ({
-      ...state,
-      importedTokens: {
-        ...state.importedTokens,
-        [action.payload.tokenId]: {
-          ...action.payload,
-          // Force imported tokens to be visible even with zero balance.
-          showZeroBalance: true,
-          balance: null,
-          isManuallyImported: true,
+    importToken: (state, action: PayloadAction<StoredTokenBalance>) => {
+      const importedTokenDetails = {
+        ...action.payload,
+        // Force imported tokens to be visible even with zero balance.
+        showZeroBalance: true,
+        isManuallyImported: true,
+      }
+
+      return {
+        ...state,
+        tokenBalances: {
+          ...state.tokenBalances,
+          [action.payload.tokenId]: importedTokenDetails,
```

Comment:
yay awesome!! so now the redux property `importedTokens` is no longer needed right? can we also remove that with a redux migration in this PR?